File: DynamicKey/AgoraDynamicKey/python/sample/education_token_builder_sample.py
# -*- coding: utf-8 -*-
__copyright__ = "Copyright (c) 2014-2022 Agora.io, Inc."

import os
import sys

sys.path.insert(0, os.path.abspath(
    os.path.join(os.path.dirname(__file__), '..')))

from src.education_token_builder import *

def main():
    app_id = "970CA35de60c44645bbae8a215061b33"
    app_certificate = "5CFd2fd1755d40ecb72977518be15d3b"
    room_uuid = "123"
    role = 1
    user_id = "2882341273"
    expiration_in_seconds = 600

    token = EducationTokenBuilder.build_room_user_token(
        app_id, app_certificate, room_uuid, user_id, role, expiration_in_seconds)
    print("Education room user token: {}".format(token))

    token = EducationTokenBuilder.build_user_token(
        app_id, app_certificate, user_id, expiration_in_seconds)
    print("Education user token: {}".format(token))

    token = EducationTokenBuilder.build_app_token(
        app_id, app_certificate, expiration_in_seconds)
    print("Education app token: {}".format(token))


if __name__ == "__main__":
    main()

File: DynamicKey/AgoraDynamicKey/dart/sample/education_token_builder_sample.dart
import 'dart:io';

import 'package:agora_token_generator/agora_token_generator.dart';

/// Generates Education room-user, user, and application Token007 tokens.
void main() {
  final appId = Platform.environment['AGORA_APP_ID'] ?? '';
  final appCertificate = Platform.environment['AGORA_APP_CERTIFICATE'] ?? '';
  const roomUuid = '123';
  const userUuid = '2882341273';
  const role = 1;
  const expireSeconds = 600;

  if (appId.isEmpty || appCertificate.isEmpty) {
    stderr.writeln(
      'Need to set environment variable AGORA_APP_ID and '
      'AGORA_APP_CERTIFICATE',
    );
    exitCode = 1;
    return;
  }

  _printToken(
    'Education room-user token',
    EducationTokenBuilder.buildRoomUserToken(
      appId: appId,
      appCertificate: appCertificate,
      roomUuid: roomUuid,
      userUuid: userUuid,
      role: role,
      expireSeconds: expireSeconds,
    ),
  );
  _printToken(
    'Education user token',
    EducationTokenBuilder.buildUserToken(
      appId: appId,
      appCertificate: appCertificate,
      userUuid: userUuid,
      expireSeconds: expireSeconds,
    ),
  );
  _printToken(
    'Education app token',
    EducationTokenBuilder.buildAppToken(
      appId: appId,
      appCertificate: appCertificate,
      expireSeconds: expireSeconds,
    ),
  );
}

/// Prints a generated token and fails when generation returns an empty value.
void _printToken(String label, String token) {
  if (token.isEmpty) {
    throw StateError('$label generation failed');
  }
  stdout.writeln('$label: $token');
}
